assets/js/components/DataBlock.js

```diff
@@ -36,193 +36,164 @@ import SourceLink from './SourceLink';
-			sparklineComponent = cloneElement( sparkline, {
-				invertChangeColor,
+	// If changeDataUnit is given, try using it as currency first, otherwise add it as suffix.
+	if ( changeDataUnit ) {
+		if ( changeDataUnit === '%' ) {
+			// Format percentage change with only 1 digit instead of the usual 2.
+			changeFormatted = numFmt( change, {
+				style: 'percent',
+				signDisplay: 'never',
+				maximumFractionDigits: 1,
 			} );
+		} else {
+			changeFormatted = numFmt( change, changeDataUnit );
 		}
+	}
 
-		let changeFormatted = change;
-
-		// If changeDataUnit is given, try using it as currency first, otherwise add it as suffix.
-		if ( changeDataUnit ) {
-			if ( changeDataUnit === '%' ) {
-				// Format percentage change with only 1 digit instead of the usual 2.
-				changeFormatted = numFmt( change, {
-					style: 'percent',
-					signDisplay: 'never',
-					maximumFractionDigits: 1,
-				} );
-			} else {
-				changeFormatted = numFmt( change, changeDataUnit );
-			}
-		}
-
-		// If period is given (requires %s placeholder), add it.
-		if ( period ) {
-			changeFormatted = sprintf( period, changeFormatted );
-		}
-
-		const datapointFormatted =
-			datapoint === undefined
-				? datapoint
-				: numFmt( datapoint, datapointUnit || undefined );
+	// If period is given (requires %s placeholder), add it.
+	if ( period ) {
+		changeFormatted = sprintf( period, changeFormatted );
+	}
 
-		return (
-			<div
-				className={ classnames(
-					'googlesitekit-data-block',
-					className,
-					`googlesitekit-data-block--${ context }`,
-					{
-						'googlesitekit-data-block--selected': selected,
-						'googlesitekit-data-block--is-gathering-data':
-							gatheringData,
-					}
-				) }
-				tabIndex={
-					'button' === context && ! gatheringData ? '0' : '-1'
-				}
-				role={ handleStatSelection && role }
-				onClick={
-					gatheringData
-						? undefined
-						: handleStatSelection && this.handleClick
-				}
-				onKeyPress={
-					gatheringData
-						? undefined
-						: handleStatSelection && this.handleKeyPress
+	const datapointFormatted = datapoint && numFmt( datapoint, datapointUnit );
+	const isButtonContext = 'button' === context;
+	const role = isButtonContext ? 'button' : '';
+
+	return (
+		<div
+			className={ classnames(
+				'googlesitekit-data-block',
+				className,
+				`googlesitekit-data-block--${ context }`,
+				{
+					'googlesitekit-data-block--selected': selected,
+					'googlesitekit-data-block--is-gathering-data':
+						gatheringData,
 				}
-				aria-disabled={ gatheringData || undefined }
-				aria-label={ handleStatSelection && title }
-				aria-pressed={ handleStatSelection && selected }
-			>
-				<div className="googlesitekit-data-block__title-datapoint-wrapper">
-					<h3
-						className="
+			) }
+			tabIndex={ isButtonContext && ! gatheringData ? '0' : '-1' }
+			role={ handleStatSelection && role }
+			onClick={ handleClick }
+			onKeyDown={ handleKeyDown }
+			aria-disabled={ gatheringData }
```

Comment:
This property value should remain the same as it was before:
```suggestion
			aria-disabled={ gatheringData || undefined }
```